src/edu/csus/ecs/pc2/clics/API202306/CLICSClarification.java

```diff
@@ -0,0 +1,246 @@
+
+import edu.csus.ecs.pc2.core.Utilities;
+import edu.csus.ecs.pc2.core.log.Log;
+import edu.csus.ecs.pc2.core.log.StaticLog;
+import edu.csus.ecs.pc2.core.model.Clarification;
+import edu.csus.ecs.pc2.core.model.ClarificationAnswer;
+import edu.csus.ecs.pc2.core.model.ClientId;
+import edu.csus.ecs.pc2.core.model.ClientType;
+import edu.csus.ecs.pc2.core.model.ContestTime;
+import edu.csus.ecs.pc2.core.model.ElementId;
+import edu.csus.ecs.pc2.core.model.Group;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.core.util.IJSONTool;
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+
+/**
+ * CLICS Clarification
+ * Contains information about a clarification
+ *
+ * @author John Buck
+ *
+ */
+public class CLICSClarification {
+
+    @JsonProperty
+    private String id;
+
+    @JsonProperty
+    private String from_team_id;
+
+    @JsonProperty
+    private String to_team_id;
+
+    @JsonProperty
+    private String [] to_team_ids;
+
+    @JsonProperty
+    private String [] to_group_ids;
+
+    @JsonProperty
+    private String reply_to_id;
+
+    @JsonProperty
+    private String problem_id;
+
+    @JsonProperty
+    private String text;
+
+    @JsonProperty
+    private String time;
+
+    @JsonProperty
+    private String contest_time;
+
+    @JsonProperty
+    private int number;
+
+    public CLICSClarification() {
+        // for jackson deserialize
+    }
+
+    public CLICSClarification(IInternalContest model, Clarification clar) {
+        this(model, clar, null);
+    }
+
+    /**
+     * Fills in the clarification properties
+     *
+     * @param model The contest
+     * @param clar The clarification
+     * @param clarAns non-null if this is an answer
+     */
+    public CLICSClarification(IInternalContest model, Clarification clar, ClarificationAnswer clarAns) {
+
+//        // use last answer if we were not handed the answer
+//        if(clarAns == null) {
+//            if (clar.isAnswered()) {
+//                // dump the answer
+//                ClarificationAnswer[] clarAnswers = clar.getClarificationAnswers();
+//                if(clarAnswers != null && clarAnswers.length > 0) {
+//                    clarAns = clarAnswers[clarAnswers.length-1];
+//                }
+//            }
+//        }
+
+
+        // pc2 specific number (really, ordinal or original question - for announcement's too!)
+        number = clar.getNumber();
+
+        // SOMEDAY change id to a original?  WTF does that mean? ordinal would make sense. -- JB
+        id = clar.getElementId().toString();
+        if (clarAns != null && clarAns.getElementId() != null) {
+            id = clarAns.getElementId().toString();
+        }
+        if (clar.getSubmitter().getClientType().equals(ClientType.Type.TEAM) && clarAns == null) {
+            from_team_id = "" + clar.getSubmitter().getClientNumber();
+        }
+        if (clarAns != null) {
+            // the request goes to a team?
```

Comment:
I think this comment means to say `//does the _answer_ go just to a team (vs. "send to all") ?`

Reply:
Yes, that seems clearer.